extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/CommonPanacheQueryImpl.java

```diff
@@ -139,12 +142,21 @@ private void checkPagination() {
         return (CommonPanacheQueryImpl<T>) this;
     }
 
+    public <T extends Entity> CommonPanacheQueryImpl<T> withReadPreference(ReadPreference readPreference) {
+        this.readPreference = readPreference;
+        return (CommonPanacheQueryImpl<T>) this;
```

Comment:
The collection should be updated/reassigned here rather than lazily.  Save some headaches down the line.

Reply:
Ah yes, this is so simple to do it this way ;)
I'll update my impl then.